lib/ansible/plugins/test/uri.py

```diff
@@ -0,0 +1,33 @@
+# (c) Ansible Project
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from urllib.parse import urlparse
+
+
+def is_uri(value, schemas=None):
+    ''' Will verify that the string passed is a valid 'uri' according to spec, if passed a list of valid schemas it will even match those '''
+    try:
+        x = urlparse(value)
+        isit = all([x.scheme, x.netloc, not schemas or x.scheme in schemas])
```

Comment:
URIs do not need an non-empty netloc. That's only a requirement for URLs as far as I understand it.